src/components/Referrals/ReferralInfoCard.tsx

```diff
@@ -0,0 +1,77 @@
+import { ARBITRUM, AVALANCHE } from "config/chains";
+import Tooltip from "../Tooltip/Tooltip";
+import StatsTooltip from "components/StatsTooltip/StatsTooltip";
+import { getUSDValue } from "./referralsHelper";
+import { useChainId } from "lib/chains";
+import { ReactNode } from "react";
+
+type Props = {
+  label: string;
+  labelTooltipText: string;
+  data: any;
+  dataKeys: Array<string>;
+  totalDataKey: string;
+  showDollar?: boolean;
+  shouldFormat?: boolean;
+  tooltipPosition?: string;
+  children?: ReactNode;
+};
+
+function ReferralInfoCard({
+  label,
+  data,
+  dataKeys = [],
+  totalDataKey,
+  showDollar = true,
+  shouldFormat = true,
+  labelTooltipText,
+  tooltipPosition = "left-bottom",
+  children,
+}: Props) {
+  const [parentKey, childKey] = dataKeys;
```

Comment:
I think it's better not to use string keys to access the data
1. it's not typed and not very intuitive, there can be bugs when updating the data or just typos
2. Hard binding on data structure, only 2nd level nesting is supported here, but for example in V2 we need to display data for V1 Arbitrum, V1 Avalanche, V2 Arbitrum, V2 Avalanche

I think it's better to keep such components as simple as possible and just format the data where it is used 

```
<ReferralInfoCard
 data={
     <StatsTooltipRow 
        label="Rebates on Avalanche"
       value={data[AVALANCHE].cumulativeStats.referrerRebates}
      >

     <StatsTooltipRow 
        label="Rebates on Arbitrum"
        value={data[ARBITRUM].cumulativeStats.referrerRebates}
      >
}

```

Reply:
Makes sense, I have refactored the ReferralsCard component. Please have a look.